src/mesh/coordinates.cxx

```diff
@@ -857,7 +857,9 @@ void Coordinates::setParallelTransform(Options* options) {
 }
 
 ParallelTransform& Coordinates::getParallelTransform() {
-  // Return a reference to the ParallelTransform object
+  if (transform == nullptr) {
+    setParallelTransform(Options::getRoot());
```

Comment:
Is this necessary and useful? `setParallelTransform` is called in the `Coordinates` constructor, except the one where all the member fields are passed in as arguments which is only used in the unit tests. If this is needed for the case when that constructor is used, could we add the `setParallelTransform(Options::getRoot())` in that constructor instead*? I worry a bit about having this conditional because I don't think the `ParallelTransform` should normally be created with `Options::getRoot()` - it should use the options that were used to create the `Mesh` that owns the `Coordinates` that own the `ParallelTransform`. So I'm worried that this might mask a bug where somehow the `ParallelTransform` didn't get set when it should have been, and make that harder to track down because the `ParallelTransform` gets set to something 'wrong' rather than being a `nullptr` that makes it obvious there's an error.

--
\* is `Options::getRoot()` always available in the unit tests? I think I remember having issues with it not being, in which case it could be a bad idea to add this call to that constructor.

Reply:
That is all entirely reasonable -- I will change this to throw instead.

`Options::getRoot()`/`Options::root()` are always available -- they just might not contain anything useful. If they are used in a test, the test/fixture should call `Options::cleanup()` on teardown.